src/lib/mdx-to-presentation.ts

```diff
@@ -3,33 +3,35 @@ import path from "node:path";
 
 import { createSlidesFolder } from "./create-slides-folder";
 import { getSlidesFromContent } from "./get-slides-from-content";
-import { getComponentsFromSlide } from "./get-components-from-slide";
-import { getImportStringsFromComponents } from "./get-import-strings-from-slide";
 
 const layout_import = `---
-layout: ../../layouts/slide-wrapper.astro
+import { SlideWrapper } from '../../layouts/SlideWrapper'
 ---
 
 `;
 
 export function mdxToPresentation(mdx: string) {
 	createSlidesFolder();
 
+	const slide_template = `<div class="slide" id="{#}"><!--SLIDE--></div>\n`;
+
+	let content = "";
 	for (const [index, slide] of getSlidesFromContent(mdx).entries()) {
-		const file_path = path.resolve(`src/pages/slides/${index + 1}.mdx`);
+		const final_slide = slide_template
+			.replace("<!--SLIDE-->", slide)
+			.replace("{#}", (index + 1).toString());
 
-		const slide_components = getComponentsFromSlide(slide);
+		content += final_slide;
+	}
 
-		const content =
-			layout_import +
-			getImportStringsFromComponents(slide_components).join("\n") +
-			"\n\n" +
-			slide;
+	const open_layout = `<SlideWrapper client:only="react">\n`;
+	const close_layout = "\n</SlideWrapper>";
+	content = layout_import + open_layout + content + close_layout;
```

Comment:
everything here becomes a single children to React

Reply:
`props.children` inside `SlideWrapper` only returns one node

Reply:
if we had multiple elements inside that `props.children`, we could use the current slide number to access the right child